include/rcutils/thread.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCUTILS__THREAD_H_
+#define RCUTILS__THREAD_H_
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#ifndef _WIN32
+# define THREAD_LOCAL_STORAGE __thread
+#else
+# define THREAD_LOCAL_STORAGE __declspec(thread)
```

Comment:
https://github.com/ros2/rcutils/blob/e52f2cfc1aac935bb0e6c35a7f3ad332a56d6d71/include/rcutils/macros.h#L33?

Reply:
It's the windows specific way of declaring a variable thread local. See https://docs.microsoft.com/en-us/cpp/cpp/thread?view=vs-2019.

Reply:
Oh, I see we already had this